src/digests/digest.lisp

```diff
@@ -177,17 +177,18 @@
              (let ((digest-size ,(if single-digest-p
                                      (second (first specs))
                                      `(etypecase state
-                                        ,@(reverse specs)))))
+                                        ,@(reverse specs))))
+                   (state-copy (copy-digest state)))
```

Comment:
If you have a need for this, then you should copy the state in the client code, rather than penalizing every client here.